vcd/resource_vcd_nsxt_edgegateway.go

```diff
@@ -584,6 +892,66 @@ func setNsxtEdgeGatewayData(edgeGateway *types.OpenAPIEdgeGateway, d *schema.Res
 	if err != nil {
 		return fmt.Errorf("error setting NSX-T Edge Gateway subnets after read: %s", err)
 	}
+	// End of 'subnet' field
+
+	// 'auto_subnet' and 'total_allocated_ip_count' fields
+	totalAllocatedIpCount, err := edgeGateway.GetAllocatedIpCount(false)
+	if err != nil {
+		return fmt.Errorf("error getting NSX-T Edge Gateway total allocated IP count: %s", err)
+	}
+	err = d.Set("total_allocated_ip_count", *totalAllocatedIpCount)
+	if err != nil {
+		return fmt.Errorf("error setting NSX-T Edge Gateway total allocated IP count: %s", err)
+	}
+
+	autoSubnets := make([]interface{}, 1)
+	for _, subnetValue := range edgeUplink.Subnets.Values {
+		oneSubnet := make(map[string]interface{})
+		oneSubnet["gateway"] = subnetValue.Gateway
+		oneSubnet["prefix_length"] = subnetValue.PrefixLength
+		oneSubnet["primary_ip"] = subnetValue.PrimaryIP
+
+		autoSubnets = append(autoSubnets, oneSubnet)
+	}
+
+	autoSubnetSet := schema.NewSet(schema.HashResource(nsxtEdgeAutoSubnetAndTotal), autoSubnets)
+	err = d.Set("auto_subnet", autoSubnetSet)
+	if err != nil {
+		return fmt.Errorf("error setting NSX-T Edge Gateway automatic subnets after read: %s", err)
+	}
+	// End of 'auto_subnet' and 'total_allocated_ip_count' fields
+
+	// 'auto_allocated_subnet' field
+	autoAllocatedSubnets := make([]interface{}, 1)
+	for _, subnetValue := range edgeUplink.Subnets.Values {
+		oneSubnet := make(map[string]interface{})
+		oneSubnet["gateway"] = subnetValue.Gateway
+		oneSubnet["prefix_length"] = subnetValue.PrefixLength
+		oneSubnet["primary_ip"] = subnetValue.PrimaryIP
+		oneSubnet["allocated_ip_count"] = *subnetValue.TotalIPCount
+
+		autoAllocatedSubnets = append(autoAllocatedSubnets, oneSubnet)
+	}
+
+	autoAllocatedSubnetSet := schema.NewSet(schema.HashResource(nsxtEdgeAutoAllocatedSubnet), autoAllocatedSubnets)
+	err = d.Set("auto_allocated_subnet", autoAllocatedSubnetSet)
+	if err != nil {
+		return fmt.Errorf("error setting NSX-T Edge Gateway auto allocated subnets after read: %s", err)
+	}
+	// End of 'auto_allocated_subnet' field
```

Comment:
If I understood, the three types are always set/computed, I guess it's because maybe one decides to switch between allocation modes and then can have real state in every mode?